tests/functional/u/use/used_before_assignment_issue5500.py

```diff
@@ -0,0 +1,9 @@
+"""used-before-assignment (E0601)"""
+def function():
+    """Assume except blocks execute if the try block returns."""
+    try:
+        return "success message"
+    except ValueError:
```

Comment:
Let's also add a tests for the finally (as mentioned in the description of the message).

And, could you rename the test file? Personally I think we should try and make the file names descriptive as well. `issue5500` doesn't tell new contributors much about what tests will be included.

Reply:
Smart! (was just trying to figure out the norms!)

Reply:
I don't think there are many norms for file names (see the regression folder...) but that doesn't stop us from starting to improve on them 😄

Reply:
> Let's also add a tests for the finally (as mentioned in the description of the message).

Good catch, this found an issue in the implementation. The finally executes before the try returns, so we need to continue to emit a message.

Reply:
Fixed in 3b28e7d.

Reply:
> was just trying to figure out the norms!

We're also doing that 😄 any ideas ? ;)